src/segment-loader.js

```diff
@@ -1389,53 +1389,61 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
   }
 
-  appendData_(segmentInfo, result) {
-    // Merge multiple video and audio segments into one and append
-    const {
-      type,
-      data
-    } = result;
-    let initSegment = result.initSegment;
+  getInitSegmentAndUpdateState_({ type, initSegment, map, playlist }) {
+    // "The EXT-X-MAP tag specifies how to obtain the Media Initialization Section
+    // (Section 3) required to parse the applicable Media Segments.  It applies to every
+    // Media Segment that appears after it in the Playlist until the next EXT-X-MAP tag
+    // or until the end of the playlist."
+    // https://tools.ietf.org/html/draft-pantos-http-live-streaming-23#section-4.3.2.5
+    if (map) {
+      const id = initSegmentId(map);
+
+      if (this.activeInitSegmentId_ === id) {
+        // don't need to re-append the init segment if the ID matches
+        return null;
+      }
 
-    if (!data || !data.byteLength) {
-      return;
+      // a map-specified init segment takes priority over any transmuxed (or otherwise
+      // obtained) init segment
+      //
+      // this also caches the init segment for later use
+      initSegment = this.initSegmentForMap(map, true).bytes;
+      this.activeInitSegmentId_ = id;
     }
 
-    if (type === 'audio' && this.audioDisabled_) {
-      return;
+    // We used to always prepend init segments for video, however, that shouldn't be
+    // necessary. Instead, we should only append on changes, similar to what we've always
+    // done for audio. This is more important (though may not be that important) for
+    // frame-by-frame appending for LHLS, simply because of the increased quantity of
+    // appends.
+    if (initSegment && this.appendInitSegment_[type]) {
+      // Make sure we track the playlist that we last used for the init segment, so that
+      // we can re-append the init segment in the event that we get data from a new
+      // playlist. Discontinuities and track changes are handled in other sections.
+      this.playlistOfLastInitSegment_[type] = playlist;
```

Comment:
do we need the whole playlist?

Reply:
Probably not necessary, but since it's only a reference, and can match precisely, it seems safest.

Reply:
hmm, ok seems fine if it's only a reference-I would be wary of caching too many things